analysis/analysis_test.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"testing"
+)
+
+// TestProcessOpenAPIDocs is a simple test that tests that filehander
+// is able to correctly process the OpenAPIdocuments in a given
+// directory
+func TestProcessOpenAPIDocs(t *testing.T) {
+	var processOASTest = []struct {
+		testName    string
+		dirPath     string
+		openAPIDocs int
+	}{
+		{"3Docs", "../incompatibility/oas-examples", 3},
```

Comment:
I think there are a couple more test cases 

Success: 
* directory with sub directory. 
* Aggregated analysis is correctly generated - You could also test those methods directly if that makes more sense.

Failure: 
* Invalid starting dir generates failure 
* Failure processing single file generates error 
    * I know you have this as a print output, wondering what is most useful to the user and how you can best test.

Reply:
- added dirWSubDir test input, test for processing subdirectory files
- continued with checking processed file num for correctness, will expand in later branches
- added falseDir test input, tests for invalid starting dir
- added dirWMalFile test input, tests for attempting to process OAS with wrong version
- error when file or directory processing is reflected in num of OAS process in analysis object